### DISCUSSION TITLE
CANNOT USE HTTP-SERVER CREATED USING NODE WITH DOCKERIZED CODE-SERVER

### DISCUSSION THREAD

**Discussion by nooobcoder** [coder/code-server#2628] [Q&A] [answered]
I am creating this simple node app to host a http server. Currently I am using code-server dockerized version to do this, when running the script i cannot access the web-server outside from it. But when i am running the same script by ssh'ing to that location from my pc's terminal, the site is accessible. Need a way to access the server by running the script from code-server.

![image](https://user-images.githubusercontent.com/50350828/105626186-469ead00-5e54-11eb-855f-f8faf75ed19d.png)
![image](https://user-images.githubusercontent.com/50350828/105626191-49999d80-5e54-11eb-84a9-f961d1cd0ba9.png)
![image](https://user-images.githubusercontent.com/50350828/105626201-4c948e00-5e54-11eb-8b3b-a5069920bdaf.png)

**Comment by code-asher** [comment, marked as answer]
You probably need to expose port 1234 on the Docker side of things.

You could also use code-server's built-in proxy to access the port by
going to /proxy/1234.

**Reply by nooobcoder** [reply]
Can we have a detailed guide on using the built-in proxy of the code-server.
Seems like modifying the port on Docker's side, recreates the container and the user data and the extensions are lost.

Also, I suggest to integrate the settings sync feature as in VSCode. The settings sync feature is greatly missed in code-server.

**Reply by nooobcoder** [reply]
I was working on a ReactJS app and I was having issues in accessing the port. @code-asher your second suggestion of using the proxy somehow worked, the page loads but no content is loaded. I am attaching the screenshot, if you can suggest me a solution to this.
Please tell me a way by which the server can redirect the request to the desired directory to access the files. Seems like if i use proxy, the original files are not available when **npm run start** done.
![image](https://user-images.githubusercontent.com/50350828/105942913-2d7d4280-6086-11eb-8614-c481c108a485.png)

The error code is this primarily: **net::ERR_ABORTED 404 (Not Found)**

**Reply by code-asher** [reply]
Here's our documentation on code-server's built-in proxy: https://github.com/cdr/code-server/blob/v3.8.0/doc/FAQ.md#how-do-i-securely-access-web-services (it's not a lot though).

Docker doesn't preserve the disk so if you want to persist files you'll need to mount the directories to your host system, for example with `-v $HOME/.local/share/code-server /home/coder/.local/share/code-server`. You might want to mount the config directory as well; see https://github.com/cdr/code-server/blob/v3.8.0/doc/install.md#docker

The issue for implementing settings sync is here: https://github.com/cdr/code-server/issues/2195

You might be running into this: https://github.com/cdr/code-server/issues/2222. We have a fix merged so hopefully we'll be able to put out a  new code-server release soon. In the meantime exposing the port in Docker is your best bet I think.
